src/test/java/org/folio/edge/patron/MainVerticleTest.java

```diff
@@ -311,6 +312,62 @@ public void testGetAccountPatronFoundIncludeAll(TestContext context) throws Exce
     assertEquals(expected, actual);
   }
 
+  @Test
+  public void testGetAccountPatronFoundIncludeAllAndLimitEqualsToOne(TestContext context) throws Exception {
+    logger.info("=== Test getAccount/includeLoans,includeCharges,includeHolds & limit=1 ===");
+
+    final Response resp = RestAssured
+      .get(String.format("/patron/account/%s?apikey=%s&includeLoans=true&includeHolds=true&includeCharges=true&limit=1&offset=0",
+        patronId, apiKey))
+      .then()
+      .statusCode(200)
+      .header(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON)
+      .extract()
+      .response();
+
+    String expected = PatronMockOkapi.getAccountWithSingleItemsJson(patronId, true, true, true);
+    String actual = resp.body().asString();
+
+    assertEquals(expected, actual);
+  }
+
+  @Test
+  public void testGetAccountPatronFoundIncludeLoansAndSortBy(TestContext context) throws Exception {
+    logger.info("=== Test getAccount/includeLoans & sortBy ===");
+
+    final Response resp = RestAssured
+      .get(String.format("/patron/account/%s?apikey=%s&includeLoans=true&includeHolds=false&includeCharges=false&sortBy=testSort&limit=10&offset=0",
+        patronId, apiKey))
+      .then()
+      .statusCode(200)
+      .header(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON)
+      .extract()
+      .response();
+
+    String expected = PatronMockOkapi.getAccountWithSortedLoans(patronId);
+    String actual = resp.body().asString();
+
+    assertEquals(expected, actual);
+  }
+
+  @Test
+  public void testGetAccountOffsetIsNegative(TestContext context) throws Exception {
+    logger.info("=== Test getAccount offset is negative ===");
+
+    int expectedStatusCode = 400;
+    final Response resp = RestAssured
+      .get(String.format("/patron/account/%s?apikey=%s&includeLoans=true&includeHolds=true&includeCharges=true&limit=1&offset=-1",
+        patronId, apiKey))
+      .then()
+      .statusCode(expectedStatusCode)
+      .extract()
+      .response();
+
+    ErrorMessage msg = ErrorMessage.fromJson(resp.body().asString());
+    assertEquals(wrongOffsetMessage, msg.message);
+    assertEquals(expectedStatusCode, msg.httpStatusCode);
```

Comment:
Why is it necessary to check the status here as well as using rest assured?

Reply:
actually, in RestAssured section, we are checking the response status, and here we are checking that the same status is present in the error message

Reply:
Why does the response body need to include a http status code?

Reply:
I did not touch the logic or erre response in edge patron. Here is the example of response when patron not found: 
![image](https://user-images.githubusercontent.com/60778653/146178273-fd544566-7f70-41a3-8b47-4949c29293ed.png)

Reply:
so response body contain code and errorMessage

Reply:
Isn't this test intended to demonstrate a failure caused by a negative offset?

Reply:
here is the example when offset is negative: 
![image](https://user-images.githubusercontent.com/60778653/146179064-af983249-5991-466e-b4f4-17eaba086908.png)

Reply:
Ok, thanks. Then previous example was a general demonstration of the structure of error messages from mod-patron

Reply:
Hmm, although the first screenshot also includes a negative offset.

Does that mean the patron is requested prior to the offset being validated?

Reply:
exactly, sorry for misleading

Reply:
investigating the prior

Reply:
yes, according to the existing logic - firstly we are looking for the patron, and here we get the 404 
https://github.com/folio-org/edge-patron/blob/a801121e726c1f10798826f68b74a3cbfc58e55f/src/main/java/org/folio/edge/patron/PatronHandler.java#L68

Reply:
@craigmcnally is that the intended behaviour (that the patron is fetched prior to the new parameters being validated)?

Reply:
looks like firstly we are making a call to mod-users /users?query=externalSystemId==id, and we get the 200 response with empty users array, and that`s why we are throwing this exception. We are not even making a call to mod-patron in this case, and offset param is used while calling to mod-patron, that`s why we see patron not found instead of offset param is invalid

Reply:
> @craigmcnally is that the intended behaviour (that the patron is fetched prior to the new parameters being validated)?

That doesn't sound right to me

Reply:
@marcjohnson-kint @craigmcnally Added validation logic for limit&offset params